common/docker/configuration.go

```diff
@@ -0,0 +1,335 @@
+package docker
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/go-connections/nat"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"pkg.world.dev/world-cli/common/config"
+)
+
+type GetDockerConfig func(cfg *config.Config) Config
+
+// DockerConfig is a configuration for a docker container
+// It contains the name of the container and a function to get the container and host config
+type Config struct {
```

Comment:
nit: I think it's better to name this `Service` to avoid ambiguity since there is a lot of "Config"

Reply:
Updated to Service, instead of Config